esp/scm/ws_dfu.ecm

```diff
@@ -832,44 +832,83 @@ ESPresponse [exceptions_inline, nil_remove] EraseHistoryResponse
     ESParray<ESPStruct History, Origin> History;
 };
 
-ESPrequest DFUFileAccessRequest
+ESPStruct DFUFileAccessRequestBase
 {
-    string Name;
     string Cluster;
     int ExpirySeconds(60);
     ESPenum FileAccessRole AccessRole;
     ESPenum SecAccessType AccessType;
-    string JobId;
     bool ReturnJsonTypeInfo(false);
     bool ReturnBinTypeInfo(false);
 };
 
-// DFUPartLocations and DFUPartCopies part of DFUFileAccessResponse below
-ESPStruct DFUPartLocations
+ESPrequest DFUFileAccessRequest
+{
+    string Name;
+    string JobId;
+    EspStruct DFUFileAccessRequestBase RequestBase;
+};
+
+ESPStruct DFUFileCopy
 {
-    int LocationIndex;
-    string Location;
+    int CopyIndex;
+    string Host;
+    string Path;
 };
 
-ESPStruct DFUPartCopies
+ESPStruct DFUFilePart
 {
     int PartIndex;
-    ESParray<string> LocationIndexes;
+    ESParray<EspStruct DFUFileCopy> Copies;
 };
 
-ESPresponse [exceptions_inline] DFUFileAccessResponse
+ESPStruct DFUFileAccessInfo
 {
     string MetaInfoBlob;
     string ExpiryTime;
-    // {NumParts, FilePartLocations, FileParts} depend on ReturnFileInfo in request
+    // {NumParts, FileParts} depend on ReturnFileInfo in request
     int NumParts;        // number of parts in logical file
-    ESParray<EspStruct DFUPartLocations> FilePartLocations;
-    ESParray<EspStruct DFUPartCopies> FileParts;
+    ESParray<EspStruct DFUFilePart> FileParts;
     
     binary RecordTypeInfoBin;   // optional
     string RecordTypeInfoJson;  // optional
 };
 
+ESPresponse [exceptions_inline] DFUFileAccessResponse
+{
+    EspStruct DFUFileAccessInfo AccessInfo;
+};
+
+ESPrequest [nil_remove] DFUFileCreateRequest
+{
+    string Name;
```

Comment:
are we going to allow the client to specify the filepart/location mapping?
We prob need a structure like this:
<FilePartsMap><FilePart><id>1</id><location>1.1.1.1></location></FilePart>....</FilePartsMap>
@jakesmith

Reply:
Yes that is what has been discussed in the HPCC-20443, which this PR should be associated with (HPCC-20164 should be closed as previously discussed/agreed).

Reply:
For the filepart/location mapping, will a string array be good enough (1st string for part 1, 2nd string for part 2, ...)? How to pass the mapping info to IDistributedFile or IFileDescriptor?

Reply:
>For the filepart/location mapping, will a string array be good enough (1st string for part 1, 2nd string for part 2, ...)?

Yes, unless @jpmcmu can think of reason needs other info too.
The paths and filenames should adhere to the info. coming back in the createfile response.

>How to pass the mapping info to IDistributedFile or IFileDescriptor?

create a group, see discussion in issue (HPCC-20443) : https://track.hpccsystems.com/browse/HPCC-20443?focusedCommentId=81520&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-81520

Reply:
@jakesmith @wangkx Yep, a file path string for each file path is all I need